src/Microsoft.DotNet.Cli.Utils/Muxer.cs

```diff
@@ -33,11 +33,24 @@ public Muxer()
             }
         }
 
+        public static string GetDataFromAppDomain(string propertyName)
+        {
+            var appDomainType = typeof(object).GetTypeInfo().Assembly?.GetType("System.AppDomain");
+            var currentDomain = appDomainType?.GetProperty("CurrentDomain")?.GetValue(null);
+            var deps = appDomainType?.GetMethod("GetData")?.Invoke(currentDomain, new[] { propertyName });
+
+            return deps as string;
+        }
+
         private bool TryResolveMuxerFromParentDirectories()
         {
-            var appBase = new FileInfo(typeof(object).GetTypeInfo().Assembly.Location);
-            var muxerDir = appBase.Directory?.Parent?.Parent?.Parent;
+            var fxDepsFile = GetDataFromAppDomain("FX_DEPS_FILE");
```

Comment:
It is empty. This property is always set to a directory or empty string. An app executing in standalone mode should not use this function, if it did it would return false or be prepared to deal with exceptions.